mk-deb.sh

```diff
@@ -125,373 +125,151 @@ function build_curvefs_python() {
     done
 }
 
-#step3 执行编译
-bazel_version=`bazel version | grep "Build label" | awk '{print $3}'`
-if [ -z ${bazel_version} ]
-then
+# step3 执行编译
+bazel_version=$(bazel version | grep "Build label" | awk '{print $3}')
+if [ -z ${bazel_version} ]; then
     echo "please install bazel 4.2.2 first"
-    exit
+    exit 1
 fi
-if [ ${bazel_version} != "4.2.2" ]
-then
-    echo "bazel version must 4.2.2"
-    echo "now version is ${bazel_version}"
-    exit
+if [ ${bazel_version} != "4.2.2" ]; then
+    echo "bazel version must be 4.2.2"
+    echo "current version is ${bazel_version}"
+    exit 1
 fi
 echo "bazel version : ${bazel_version}"
 
-
 # check gcc version, gcc version must >= 4.8.5
-gcc_version_major=`gcc -dumpversion | awk -F'.' '{print $1}'`
-gcc_version_minor=`gcc -dumpversion | awk -F'.' '{print $2}'`
-gcc_version_pathlevel=`gcc -dumpversion | awk -F'.' '{print $3}'`
-if [ ${gcc_version_major} -lt 4 ]
-then
-    echo "gcc version must >= 4.8.5, current version is "`gcc -dumpversion`
-    exit
+gcc_version_major=$(gcc -dumpversion | awk -F'.' '{print $1}')
+gcc_version_minor=$(gcc -dumpversion | awk -F'.' '{print $2}')
+gcc_version_pathlevel=$(gcc -dumpversion | awk -F'.' '{print $3}')
+if [ ${gcc_version_major} -lt 4 ]; then
+    echo "gcc version must >= 4.8.5, current version is "$(gcc -dumpversion)
+    exit 1
 fi
 
-if [[ ${gcc_version_major} -eq 4 ]] && [[ ${gcc_version_minor} -lt 8 ]]
-then
-    echo "gcc version must >= 4.8.5, current version is "`gcc -dumpversion`
-    exit
+if [[ ${gcc_version_major} -eq 4 ]] && [[ ${gcc_version_minor} -lt 8 ]]; then
+    echo "gcc version must >= 4.8.5, current version is "$(gcc -dumpversion)
+    exit 1
 fi
 
-if  [[ ${gcc_version_major} -eq 4 ]] && [[ ${gcc_version_minor} -eq 8 ]] && [[ ${gcc_version_pathlevel} -lt 5 ]]
-then
-    echo "gcc version must >= 4.8.5, current version is "`gcc -dumpversion`
-    exit
+if [[ ${gcc_version_major} -eq 4 ]] && [[ ${gcc_version_minor} -eq 8 ]] && [[ ${gcc_version_pathlevel} -lt 5 ]]; then
+    echo "gcc version must >= 4.8.5, current version is "$(gcc -dumpversion)
+    exit 1
 fi
-echo "gcc version : "`gcc -dumpversion`
-echo "start compile"
+echo "gcc version : "$(gcc -dumpversion)
 
-cd ${dir}/thirdparties/etcdclient
-make clean
-make all
-if [ $? -ne 0 ]
-then
-    echo "make etcd client failed"
-    exit
-fi
-cd ${dir}
+echo "start compiling"
+
+cd ${dir}/thirdparties/etcdclient &&
+    make clean &&
+    make all &&
+    cd $OLDPWD
+
+cd ${dir}/thirdparties/memcache &&
```

Comment:
Do you mean we can skip building these two components? During my testing, if i skipped these two components, the build cannot proceed due to missing headers.

Reply:
> Do you mean we can skip building these two components? During my testing, if i skipped these two components, the build cannot proceed due to missing headers.

yep, This https://github.com/opencurve/curve/pull/2415 solves the memcached problem.

Reply:
For rocksdb, you can check where it is dependent and see if it is necessary.

Reply:
> yep, This #2415 solves the memcached problem.

So do you mean with #2415 , we don't need to build memcached? But on a clean build machine, there is no `.so` libraries. It seems to me that they still need to be built.

Reply:
> For rocksdb, you can check where it is dependent and see if it is necessary.

I think yes. rocksdb headers are missing on my first attempt. I will double-check to confirm it later.

Reply:
> > For rocksdb, you can check where it is dependent and see if it is necessary.
> 
> I think yes. I remembered rocksdb headers were missing on my first attempt. I will double-check to confirm it later.

I checked and found that the problem should be something like this, here is the build...
```bash
bazel build ... --copt -DHAVE_ZLIB=1 --compilation_mode=dbg -s --define=with_glog=true \
         --define=libunwind=true --copt -DGFLAGS_NS=google --copt \
         -Wno-error=format-security --copt -DUSE_BTHREAD_MUTEX --copt -DCURVEVERSION=${curve_version} \
         --linkopt -L/usr/local/lib ${bazelflags}
```
This will compile curvefs.
It is recommended to change to
```bash
make build stor=bs release=0 dep=1
```
Compile release:
```bash
make build stor=bs release=1 dep=1
```

Reply:
Sure, that will be much better!

Reply:
fixed